src/Runner.Common/JobServer.cs

```diff
@@ -56,6 +67,40 @@ public async Task ConnectAsync(VssConnection jobConnection)
             _hasConnection = true;
         }
 
+        private async Task CheckHealthEndpointsAsync()
+        {
+            try
+            {
+                Trace.Info("Requesting Actions Service health endpoint status");
+                // Construct the HttpClient object directly, without going through GetClient<T>
+                // as we haven't properly connected to the server at this point
+                var baseUri = new Uri(_connection.Uri.GetLeftPart(UriPartial.Authority));
+                var client = new HealthHttpClient(baseUri, _connection.InnerHandler, false);
+
+                // Call the _apis/health endpoint
+                await client.GetHealthAsync();
+
+                Trace.Info("Requesting Github API endpoint status");
+                // This is a dotcom public API... just call it directly
+                using (var httpClientHandler = HostContext.CreateHttpClientHandler())
+                using (var gitHubClient = new HttpClient(httpClientHandler))
+                {
+                    gitHubClient.BaseAddress = new Uri("https://api.github.com");
```

Comment:
i would not set the `BaseAddress`, but `gitHubClient.GetAsync("https://api.github.com")`